selfhost/typechecker.jakt

```diff
@@ -3957,6 +3957,63 @@ struct Typechecker {
             let type_id = .find_or_add_type_id(Type::GenericInstance(id: array_struct_id, args: [inner_type_id]))
             yield CheckedExpression::JaktArray(vals, repeat, span, type_id)
         }
+        JaktDictionary(values, span) => {
+            let VOID_TYPE_ID = builtin(BuiltinType::Void)
+            mut key_type_id = unknown_type_id()
+            mut value_type_id = unknown_type_id()
+            mut vals: [(CheckedExpression, CheckedExpression)] = []
+            let dictionary_struct_id = .find_struct_in_prelude("Dictionary")
+
+            for kv in values.iterator() {
+                let key = kv.0
+                let value = kv.1
+
+                let checked_key = .typecheck_expression(key, scope_id, safety_mode)
+                let current_key_type_id = expression_type(checked_key)
+
+                let checked_value = .typecheck_expression(value, scope_id, safety_mode)
+                let current_value_type_id = expression_type(checked_value)
+
+                if key_type_id.equals(unknown_type_id()) and value_type_id.equals(unknown_type_id()) {
+                    if current_key_type_id.equals(VOID_TYPE_ID) {
+                        .error("cannot create a dictionary with keys of type void", span)
```

Comment:
Maybe also fix the capitalization of the error message, you see Andreas (see 299f73379d5ffd6d840a819ba831e0b193f566e5)  doing this in many of his commits, so you'd have to change the test-files and the rust version also (see my PR for reference)